src/Decco.re

```diff
@@ -72,6 +72,24 @@ let boolFromJson = (j) =>
 let unitToJson = () => Js.Json.number(0.0);
 let unitFromJson = (_) => Belt.Result.Ok(());
 
+let jsDateToJson = (v: Js.Date.t) =>
+  Js.Date.toJSONUnsafe(v) |> Js.Json.string;
+let jsDateFromJson = j =>
+  try(
+    {
+      let date =
+        Js.Json.decodeString(j)->Belt.Option.getExn->Js.Date.fromString;
+      if (Js.Float.isNaN(Js.Date.getTime(date))) {
+        failwith("NaN");
```

Comment:
This should be `Belt.Result.Error` instead, using Result would make the error propagate to the parent and won't crash at runtime.